src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/GraphSymbolUtils.cs

```diff
@@ -143,18 +143,161 @@ private static IEnumerable<INamedTypeSymbol> GetAllViewTypesFromPXGraphOrPXGraph
+            }
+
+            void AddDelegatesFromGraphExtension(Dictionary<string, (MethodDeclarationSyntax, IMethodSymbol)> delegates, ITypeSymbol graphExt)
+            {
+                IEnumerable<(MethodDeclarationSyntax, IMethodSymbol)> dels = graphExt.GetViewDelegatesFromGraphOrGraphExtension(views, pxContext, cancellation);
+
+                foreach ((MethodDeclarationSyntax, IMethodSymbol) d in dels)
+                {
+                    delegates[d.Item2.Name] = d;
+                }
+            }
+        }
+
+        private static IEnumerable<T> GetViewInfoFromGraphExtension<T>(ITypeSymbol graphExtension, PXContext pxContext,
+            Action<Dictionary<string, T>, ITypeSymbol> addGraphViewInfo,
+            Action<Dictionary<string, T>, ITypeSymbol> addGraphExtensionViewInfo)
+        {
+            IEnumerable<T> empty = Enumerable.Empty<T>();
+
+            if (!graphExtension.InheritsFrom(pxContext.PXGraphExtensionType) || !graphExtension.BaseType.IsGenericType)
+            {
+                return empty;
+            }
+
+            INamedTypeSymbol baseType = graphExtension.BaseType;
+            Dictionary<string, T> infoByView = new Dictionary<string, T>(StringComparer.OrdinalIgnoreCase);
+
+            for (int i = baseType.TypeArguments.Length - 1; i >= 0; i--)
+            {
+                ITypeSymbol argType = baseType.TypeArguments[i];
+
+                if (i == baseType.TypeArguments.Length - 1)
+                {
+                    if (!argType.IsPXGraph(pxContext))
+                    {
+                        return empty;
+                    }
+
+                    addGraphViewInfo(infoByView, argType);
+                }
+                else
+                {
+                    if (!argType.IsPXGraphExtension(pxContext))
+                    {
+                        return empty;
+                    }
+
+                    addGraphExtensionViewInfo(infoByView, argType);
+                }
+            }
+
+            addGraphExtensionViewInfo(infoByView, graphExtension);
+
+            return infoByView.Values;
+        }
+
+        private static IEnumerable<(MethodDeclarationSyntax Node, IMethodSymbol Symbol)> GetViewDelegatesFromGraphOrGraphExtension(
+            this ITypeSymbol graphOrExtension, IEnumerable<ISymbol> views, PXContext pxContext, CancellationToken cancellation)
+        {
+            foreach (ISymbol member in graphOrExtension.GetMembers())
+            {
+                cancellation.ThrowIfCancellationRequested();
+
+                if (!(member is IMethodSymbol method))
+                {
+                    continue;
+                }
+
+                bool correctName = views.Any(v => v.Name.Equals(member.Name, StringComparison.OrdinalIgnoreCase));
+                if (!correctName)
+                {
+                    continue;
+                }
+
+                bool correctReturnType = pxContext.SystemTypes.IEnumerable.Equals(method.ReturnType);
```

Comment:
@metallmonster this doesn't consider the `IEnumerable<T>` return type. This must be handled.

Reply:
@metallmonster @SENya1990 I took a look on the platform code and it seems that a delegate can have only two signatures:
1. `IEnumerable` return type with any number of parameters, `ref` parameters are not allowed (`PXSelectDelegate`)
2. `void` return type with any number of parameters, but if any, they must be by `ref` (`PXPrepareDelegate`). This signature is used for parameters transformation.

Please note that a delegate can be either instance or static, public or non-public.

Generic `IEnumerable<T>` is not supported.

See `PXGraph.ProcessFields` method and  `PXView` constructor for more details.

Reply:
@dropsonic yes, you're right - checked it and it works only for IEnumerable.